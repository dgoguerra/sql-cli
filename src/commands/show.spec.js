const { getTestKnex, runCli } = require("../../test/utils");
const { hydrateKnex } = require("../knex/knex");

describe("show", () => {
  let knex;
  let connUri;

  beforeAll(async () => {
    knex = hydrateKnex(getTestKnex());
    connUri = knex.getUri();

    await knex.schema.createTable("table_1", (t) => {
      t.increments("id");
      t.integer("field_1");
      t.string("field_2").defaultTo("default text");
      t.string("field_3").references("table_1.field_2");
      t.timestamps();
      t.index(["field_2", "field_1"]);
    });
  });

  it("can show table", async () => {
    const output = await runCli(`show ${connUri}/table_1`);
    expect(output).toMatchSnapshot();
  });
});
