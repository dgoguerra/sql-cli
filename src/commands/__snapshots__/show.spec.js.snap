// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show can show table 1`] = `
"column     type         nullable default      foreign key
id         integer      false
field_1    integer      true
field_2    varchar(255) true     default text
field_3    varchar(255) true                  table_1.field_2
created_at datetime     true
updated_at datetime     true

index                         algorithm unique columns
table_1_field_2_field_1_index unknown   false  field_2,field_1
"
`;
